rclpy/rclpy/node.py

```diff
@@ -349,16 +350,23 @@ def create_publisher(
         msg_type,
         topic: str,
         *,
-        qos_profile: QoSProfile = qos_profile_default
+        qos_profile: QoSProfile = qos_profile_default,
+        callback_group: CallbackGroup = None,
+        event_callbacks: PublisherEventCallbacks = None,
```

Comment:
If we have this class that tracks the groups the PublisherEventCallbacks does it makes sense to just add the CallbackGroup to that constructor instead of to the `create_publisher` function? Thoughts?

Reply:
Discussed offline - going to keep it mirrored to Subscription interface. In future we may want to consider having EventCallbacks use a separate CallbackGroup than the Subscription message callback, in which case both Subscription and Publisher will need that update update